### ISSUE TITLE
Values with single quotes break the Edit Badges functionality

### ISSUE THREAD

**Issue by aqualectrix** [saltcorn/saltcorn#2567]
**Describe the bug**
Any badge text that includes a single quote (acting as an apostrophe) cannot be added or removed via the Edit Badges view.

This seems to be because in the produced HTML, the badge values are surrounded by single quotes when passed as values to functions -- and the single quotes inside the actual values are un-escaped, so they terminate the value prematurely and cause all sorts of parsing problems when the javascript is triggered.

**To Reproduce**
1. Create the TestA table with a single String field called "StringA"
2. Create the TestB table with a single String field called "StringB"
3. Create the TestAB table with the KeyA field (Key to TestA) and the KeyB field (Key to TestB)
4. On TestA, create a view, ABEditBadges, with the Edit Badges view type. The Relation should be TestAB.keya->keyb->stringb.
5. On TestA, create a view, EditA. Duplicate the StringA column setup and change it like this:
  - **Text**: Change the Text to display to "Bs"; make Label for Field blank
  - **Field**: Delete the field component and add a View component in its place. The View to show is ABBadges.
6. Add a record to TestA with this StringA values:
  - One
7. Add records to TestB with these StringB values:
  - Red
  - Blue's
8. Open the EditA view and add `?id=1` to the end of the URL. Navigate to that URL to edit the record.
9. In the Edit Badge dropdown, choose "Red". A "Red" badge appears.
10. **In the Edit Badge dropdown, choose "Blue's". No badge appears.** 

11. In the TestAB table, add a row with KeyA = One and KeyB = Blue's
12. Open the EditA view (with `?id=1`) again. Observe that the badge with the "Blue's" value correctly appears.
13. Click the X on the "Red" badge. The "Red" badge disappears.
14. Click the X on the "Blue's" badge. The "Blue's" badge remains.

**Expected behavior**
Badge values with single-quotes used as apostrophes should be valid -- they should be addable and removable via the Edit Badges view.

**Comment by aqualectrix** [comment]
The add onclick for "Blue's" looks like this:
```
<a class="dropdown-item" onclick="set_add_badge_bs43727('Blue's')">Blue's</a>
```

The remove onclick for "Blue's" looks like this:
```
<a onclick="(function(that){view_post('ABBadges', 'remove', {id:'1', value: 'Blue's'}, etc.
```

You can see that because of the unescaped single quotes in the badge display values, we suddenly have strings ending too soon and javascript parsing getting very confused indeed.

**Comment by christianhugoch** [comment]
I've added a pr: https://github.com/saltcorn/badges/pull/8/files

**Comment by glutamate** [comment]
Fixed in badges 0.3.0
